plugins/utils/yanky.nix

```
@@ -174,80 +302,50 @@ in
+            p = "require('yanky.telescope.mapping').put('p')";
+            P = "require('yanky.telescope.mapping').put('P')";
+            gp = "require('yanky.telescope.mapping').put('gp')";
+            gP = "require('yanky.telescope.mapping').put('gP')";
+            d = "require('yanky.telescope.mapping').delete()";
+            r = "require('yanky.telescope.mapping').set_register(require('yanky.utils').get_default_register())";
+          };
+        };
+      };
+    };
+  };
 
-      plugins.telescope.enable = true;
-      plugins.telescope.enabledExtensions = optional cfg.picker.telescope.enable "yank_history";
-
-      extraPlugins = [
-        cfg.package
-      ] ++ (optional (cfg.ring.storage == "sqlite") pkgs.vimPlugins.sqlite-lua);
-
-      extraConfigLua =
-        let
-          setupOptions =
-            with cfg;
-            {
-              ring = {
-                history_length = ring.historyLength;
-                inherit (ring) storage;
-                storage_path = ring.storagePath;
-                sync_with_numbered_registers = ring.syncWithNumberedRegisters;
-                cancel_event = ring.cancelEvent;
-                ignore_registers = ring.ignoreRegisters;
-                update_register_on_cycle = ring.updateRegisterOnCycle;
-              };
-              picker = {
-                select = {
-                  action =
-                    if isString picker.select.action then
-                      helpers.mkRaw "require('yanky.picker').actions.${picker.select.action}"
-                    else
-                      picker.select.action;
-                };
-                telescope = {
-                  use_default_mappings = picker.telescope.useDefaultMappings;
-                  mappings = helpers.ifNonNull' picker.telescope.mappings (
-                    mapAttrs (
-                      mode: mappings: mapAttrs (key: action: helpers.mkRaw "__yanky_telescope_mapping.${action}") mappings
-                    ) picker.telescope.mappings
-                  );
-                };
-              };
-              system_clipboard = {
-                sync_with_ring = systemClipboard.syncWithRing;
-              };
-              highlight = {
-                on_put = highlight.onPut;
-                on_yank = highlight.onYank;
-                inherit (highlight) timer;
-              };
-              preserve_cursor_position = {
-                inherit (preserveCursorPosition) enabled;
-              };
-              textobj = {
-                inherit (textobj) enabled;
-              };
-            }
-            // cfg.extraOptions;
-        in
-        (optionalString thereAreSomeTelescopeMappings "local __yanky_telescope_mapping = require('yanky.telescope.mapping')")
-        + ''
-          require('yanky').setup(${helpers.toLuaObject setupOptions})
-        '';
+  extraOptions = {
+    enableTelescopeExtension = mkEnableOption "the `yank_history` telescope picker.";
```

```suggestion
    enableTelescope = mkEnableOption "the `yank_history` telescope picker.";
```

See [existing examples](https://github.com/search?q=repo%3Anix-community%2Fnixvim%20enableTelescope&type=code)

---

Indeed, done.